### ISSUE TITLE
App never detects accessibility permission is given

### ISSUE THREAD

**Issue by arijitroy454** [maforn/TimedShutdownAndroid#24]
Randomly asks to enable accessibility permission even when it is already enabled and app is working perfectly.

![photo_2024-09-24_04-24-41](https://github.com/user-attachments/assets/32f16e71-cbfd-4054-96b4-3c00ed13f852)
![photo_2024-09-24_04-24-37](https://github.com/user-attachments/assets/2bc02b44-bdad-44ae-ae46-4842a9d972b8)

**Comment by arijitroy454** [comment]
Android 13, infinix zero 30

**Comment by arijitroy454** [comment]
Could bypass this just by adding a "never show" option to the warnings.

**Comment by arijitroy454** [comment]
Nevermind, its not random. the app just never detects that it has the permissions after the app is started for the first time. Everything works after clicking "cancel", but the warning is very annoying.

**Comment by maforn** [comment]
This is quite a strange issue, as you can see from the code it should only show you the alert dialog when the accessibility service is not enabled. Can you specify in which cases this happens?
```Java
if (!AccessibilityService.isAccessibilityServiceEnabled(requireContext(), AccessibilityService.class)) {
      AccessibilityService.requireAccessibility(getContext());
}

**Comment by arijitroy454** [comment]
Hello, it seems that the way to reproduce the issue (on my device) is to open the app and close it by clearing ram multiple times. after 3-4 times the app starts showing the warning.

here is a video of me reproducing the issue ->

https://github.com/user-attachments/assets/80411db5-d68f-49ce-8824-2c351b83ea08

after i hit cancel (to remove the warning), everything works perfectly and i can even use the app to shutdown/restart.

now at this stage, two things can happen.

1. if i close the app again by clearing ram, the next time i open the app, the warning will show without a fail.
2. if i close the app by swiping it, the warning won't show the next time i open it.

**Comment by maforn** [comment]
You are right, this happened to me as well when I followed your instructions. This is crazy, I have no clue why this is happening. As soon as I have enough free time I'll plug my phone to my computer, attach the adb and debug it. I'll keep you updated on the progress.

**Comment by maforn** [comment]
So, I've debugged quite a bit and found a problem in the `getEnabledAccessibilityServiceList()` function from the Android core. It's obviously not something I can fix by myself, as it's a bug in an Android core library. I've changed the code to use another function, which is `Settings.Secure.getString()` and it looks like it does work without any problem now. I'll push this improvement as soon as I manage to finish all the other requests that I got, thank you for bringing this to my attention.
